docs/false_positives.md

```diff
@@ -0,0 +1,243 @@
+
+Some functions only work on a set of possible input values. This precondition
+is unknown to the analyzer unless it is expressed explicitly in the code.
+
+Let us consider the following false positive:
+
+```cpp
+int f(MyEnum Val) {
+  int x = 0;
+  switch (Val) {
+    case MyEnumA: x = 1; break;
+    case MyEnumB: x = 5; break;
+  }
+  return 5/x; // Division by zero when Val == MyEnumC.
+}
+```
+
+It can be rewritten to as the following to suppress the warning:
+
+```cpp
+int f(MyEnum Val) {
+  int x = 0;
+  switch (Val) {
+    case MyEnumA: x = 1; break;
+    case MyEnumB: x = 5; break;
+    default: assert(false); break;
+  }
+  return 5/x; // No warning.
+}
+```
+
+Other macros or builtins expressing unreachable code may be used.
+Note that the rewritten code is also safer, since debug builds now check for
+more precondition violations. 
+
+#### <a name="prefer-standard-functions"></a> Prefer standard functions
+
+The analyzer models the behavior of some standard functions but it has
+no knowledge about the semantics of custom made functions declared in a
+library that compiled separately (or in another translation unit).
+
+Let us consider the following false positive:
+
+```cpp
+int f() {
+  const char *Str = "MyStr";
+  int Val = 0;
+  ...
+  if (mystrlen(Str)) {
+    Val = 1;
+  }
+  return 5 / Val; // Division by zero.
+}
+```
+
+It can be rewritten to as the following to suppress the warning:
+
+```cpp
+int f() {
+  const char *Str = "MyStr";
+  int Val = 0;
+  ...
+  if (strlen(Str)) {
+    Val = 1;
+  }
+  return 5 / Val; // No warning.
+}
+```
+
+The result is easier to read for other developers who might not be familiar
+with the custom version of the function. The standard library functions also
+tend to be faster and more correct than custom solutions.
+
+#### <a name="do-not-turn-off-code-checks"></a> Do not turn off code checks
+
+Checks in the static analyzer are usually not just reporting issues but
+also help modelling language constructs and library functions. You should never
+turn off checks from the core package.
+
+#### <a name="suppress-specific-dead-store-warnings"></a> Suppress specific dead store warnings
+
+How to suppress a specific dead store warning and more useful tips can be found
+[here](https://clang-analyzer.llvm.org/faq.html).
+
+#### <a name="alternative-implementations"></a> Alternative implementations
+
+If there are huge source of false positives due to the analyzer can not model a
+function properly you could either disable the analyzer to analyze that
+function or provide an alternative implementation that is easier to model
+(i.e.: implementation without bitwise operation tricks). This can be achieved
+using the `__clang_analyzer__` macro.
+
+### <a name="syntax-based-checks"></a> Syntax based checks
+
+Clang tidy has lots of useful syntax based checks. Some of these checks
+find bugprone code snippets. When these snippets are intentional, usually
```

Comment:
`bugprone` is the name of the check "family", but isn't a word when written like this. `Bug-prone` is more English.